ui/ui-components/__tests__/components/profilePreview.tsx

```diff
@@ -0,0 +1,112 @@
+import { render, act } from "@testing-library/react";
+import ProfilePreview from "../../components/property/profilePreview";
+import { ErrorHandler } from "../../utils/errorHandler";
+
+jest.mock("next/router", () => {
+  const nextRouter = jest.requireActual("next/router");
+  nextRouter.useRouter = jest.fn(() => ({
+    route: "/",
+    query: {
+      profileId: "asdf",
+    },
+    replace: jest.fn(),
+    push: jest.fn(),
+  }));
+  return nextRouter;
+});
+
+const execApi = jest.fn(() => {
+  return Promise.resolve({});
+});
+jest.mock("../../hooks/useApi", () => ({
+  useApi: () => ({
+    execApi,
+  }),
+}));
+
+describe("property / profilePreview", () => {
+  afterAll(() => {
+    jest.unmock("next/router");
+    jest.unmock("../../hooks/useApi");
+  });
+  afterEach(() => {
+    execApi.mockReset();
+  });
+
+  it("should render with no profile preview", () => {
+    expect(
+      render(
+        <ProfilePreview
+          errorHandler={new ErrorHandler()}
+          localFilters={{}}
+          property={{
+            id: "test-id",
+            key: "test-key",
+            type: "test-type",
+            options: {},
+          }}
+        />
+      ).container
+    ).toMatchSnapshot();
+  });
+
+  it("should render", async () => {
+    execApi.mockImplementation(() => {
+      return Promise.resolve({
+        profile: {
+          id: "asdf",
+          properties: {
+            date: { type: "date", values: [1631127518020] },
+          },
+        },
+      });
+    });
+    let profile: ReturnType<typeof render>;
+    await act(async () => {
+      profile = render(
+        <ProfilePreview
+          errorHandler={new ErrorHandler()}
+          localFilters={{}}
+          property={{
+            id: "test-id",
+            key: "test-key",
+            type: "test-type",
+            options: {},
+          }}
+        />
+      );
+      await new Promise((res) => setTimeout(res, 2000));
+    });
+    expect(profile.container).toMatchSnapshot();
```

Comment:
Newbie question: Why is snapshot testing preferred vs an explicit expectation?

Reply:
Querying into the rendered element or doing text matching is fine, but I have to reasons:
1. When making changes to the component you'll see structural changes reflected in the snapshot. _Ideally_ you'll be able to determine that you wanted those changes.
2. It's a super easy way to test and keep test data/direct comparisons out of the test code itself. You do have to eyeball it the first run, but then it's pretty consistent.